lib/si-events-rs/src/audit_log.rs

```diff
@@ -1,53 +1,36 @@
-use chrono::{DateTime, Utc};
+use chrono::Utc;
 use serde::{Deserialize, Serialize};
-use strum::{AsRefStr, Display, EnumIter, EnumString};
 
-use crate::Actor;
+use v1::AuditLogV1;
+use v2::{AuditLogKindV2, AuditLogKindV2Discriminants, AuditLogV2};
 
+use crate::{Actor, ChangeSetId};
+
+mod v1;
+mod v2;
+
+pub type AuditLogKind = AuditLogKindV2;
+pub type AuditLogKindDiscriminants = AuditLogKindV2Discriminants;
+
+// TODO(nick): switch to something like "naxum-api-types" crate to avoid sizing issues.
+#[allow(clippy::large_enum_variant)]
```

Comment:
Well! I forgot about this scenario! I wonder if `Box`ing each variants member would not be too catastrophic for the API ergonomics. Reminds me...I bet there are some other request/reponse types which will suffer from this. And it's a little worse because these types get larger and larger over time when we are passing them around like they are pretty small. Anyway, very interesting find!

Reply:
Yeah this will be a pain. Frankly, I'm considering using that `naxum-api-types` crate I made from your Rebase Request versioning code that I ported over. I am a bit concerned about exploding enums here.